c++/src/kj/compat/tls.c++

```diff
@@ -315,7 +327,7 @@ private:
             // According to documentation we shouldn't get here, because our BIO never returns an
             // "error". But in practice we do get here sometimes when the peer disconnects
             // prematurely.
-            KJ_FAIL_ASSERT("TLS protocol error");
+            return KJ_EXCEPTION(DISCONNECTED, "SSL unable to continue I/O");
```

Comment:
Out of curiosity, did you actually see this error manifest in some testing? Or is this change based on the comment above (which, to be fair, does seem to suggest that DISCONNECTED is correct here).

Reply:
This one I actually did see in testing when I let my client side disappear---I was confused when I seemed to still need the KJ_EXPECT_LOG despite adding the exemption. IIRC, most disconnections will surface as ECONNRESET/ECONNREFUSED during socket operations. I believe SSL will only politely close the socket with a closure alert if there was a protocol violation.